DEPS

```diff
@@ -151,6 +155,12 @@ hooks = [
     'condition': 'checkout_win',
     'action': ['python3', 'build/util/update_midl_files.py']
   },
+  {
+    'name': 'build_libdmg_hfsplus',
+    'pattern': '.',
+    "condition": 'checkout_mac and host_os != "mac"',
+    'action': ['build/mac/cross-compile/build-libdmg-hfsplus.py', 'third_party/libdmg-hfsplus']
+  },
```

Comment:
I understand that this will simplify things a lot for you while working on mac installers, but I don't think we need this be enabled by default when targetting `mac` on `linux`.

Can we make all this stuff be opt-in with a gclient variable? This way you can enable it in `.gclient` via `custom_vars` section, but for other cross-compile scenarios we won't be bound with all these additional things that won't be used 99% of the time.

I'd say we should support this via `.env` variable. When enabled, it may add additional var into `.gclient` and it should enable special path manipulations you're doing in `config.js`.

Reply:
While I understand your argument, there is one subtle point that I feel it does not adequately take into account.

Your argument would make sense if there were 1,000 people or CI jobs that cross-compile. Using your number of 99%, only 10 people would then enable the `.env` or `.gclient` toggle, leaving the other 990 people/bots untouched.

But, at the moment, the total number of people and CI jobs that cross-compile is 1 - me. 100% of these people are going to set the toggle.

Yes, this number will likely be higher in the future. But even then, the number of people and jobs that do not enable the toggle will be low.

I can see a point of a `.gclient` `custom_var` that only checks out / builds the `libdmg-hfsplus` tool when enabled. That tool is only required for creating DMGs, and it incurs a dependency (CMake) that not all devs may have. If a dev just wants to cross-compile Brave, but not create a DMG, then the `custom_var` makes their life easier. Furthermore, the `custom_var` is easy to implement.

A `.env` variable that also controls the `PATH` and `PYTHONPATH` modifications however seems over the top for me, given the small numbers I mentioned above and the small effect of the path modifications.  Specifically, the path modifications do not change but only add functionality that is otherwise not available on Linux. There is no risk of Linux accidentally calling code that it expects to behave as on Linux, but it actually behaving as on macOS.

In light of this, is it OK with you if I only add the `.gclient` variable, but leave the path logic unchanged? As I said, I think it would be overkill to guard it by an entry in .env.

Reply:
An alternative solution I can think of, which however requires a patch, is to change upstream's `mac_ver` to read an environment variable, say `MACOS_VERSION`. Then anybody who cross-compiles can just set this to the version of macOS host version. If the env. var isn't set, then the implementation could fall back to upstream's normal implementation.

Reply:
Another alternative might be to pretend to the build process that the file `/System/Library/CoreServices/SystemVersion.plist` exists. This is where Python's `mac_ver()` reads from ([source](https://github.com/python/cpython/blob/cecaceea31f32f01b5617989e3dc8b2077f53f89/Lib/platform.py#L458C5-L458C58)).

Reply:
> Code signing, where upstream calls `mac_ver` in [macos_version()](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/chrome/installer/mac/signing/commands.py#125). This in turn gets called by upstream in [sign_part(...)](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/chrome/installer/mac/signing/signing.py#148), via [_linker_signed_arm64_needs_force(...)](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/chrome/installer/mac/signing/signing.py#64).

if it's the only place, then I don't see any issues with preparing this value first via `run_on_macos.py` as part of hooks and then patch `macos_version()` (via `brave/chromium_src` or a raw patch if `brave/chromium_src` won't work).

Reply:
It might not be possible to fetch the value during hooks because the macOS host might not be reachable then.

What do you think of 3fcefc805f364998a9e6379d0ba07bc788111a4b? It requires cross-compilers to set environment variable `MACOS_VERSION`. It uses a patch but keeps the `PYTHONPATH` unchanged.

Reply:
is `macos_version()` required to work as a part of `//brave/build/mac/sign_app.sh`? or is it used by other GN steps that we don't own?

if we own the step(s) entirely, it would be nice to generate it automatically and only require `MACOS_HOST` for the whole thing to work, but up to you.